api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -665,6 +665,11 @@ message HttpConnectionManager {
   // information about internal/external addresses.
   InternalAddressConfig internal_address_config = 25;
 
+  // If set, Envoy will not append the remote address to the
+  // :ref:`config_http_conn_man_headers_x-envoy-overloaded` HTTP header in the scenario where
+  // the Overload Manager has been triggered
+  bool skip_overload_append = 57;
```

Comment:
I am thinking the reverse way, like this can be a `append_local_overload_header`, since the current users are not expecting this header by default and setting this to ture will append it.

Reply:
Looks the router is adding the overload header by default with a config `suppress_envoy_headers_` to disable it.
Glad to hear @adisuissa's opinion.

Reply:
I agree with @botengyao that this should be an opt-in feature, i.e., Envoy should not send that info unless it is enabled.

Maybe I'm missing something, but where is Envoy adding the "remote address to the x-envoy-overloaded HTTP header"? Is there a test.
I also think that if this is the desired outcome, then a different header should be used, maybe `x-envoy-upstream-overloaded` as this is different than an Envoy overload.

RE the suppress-envoy-headers, as this is an Envoy header, it should behave as any other Envoy header, and if suppression is enabled, it should not be added.
I suggest adding an integration test with the feature enabled to test this behavior, and also update the docs to document it.

Reply:
Thank you for working on this @leosarra. I agree with folks above about this being opt-in. I also suggest that it has its own header value vs leveraging the existing one to avoid conflating issues, impacting monitoring, etc.